### ISSUE TITLE
NFS client install returning a 404

### ISSUE THREAD

**Issue by affixalex** [maier/vagrant-alpine#15]
Firstly, thanks for this! I was excited, and I'm a little bummed to be getting this error now.

```
fetch http://repos.dfw.lax-noc.com/alpine/v3.3/main/x86_64/APKINDEX.tar.gz
ERROR: http://repos.dfw.lax-noc.com/alpine/v3.3/main: No such file or directory
v3.3.1-58-ga22d4a0 [http://repos.dfw.lax-noc.com/alpine/v3.3/main]
1 errors; 5314 distinct packages available
```

**Comment by maier** [comment]
Sounds like an issue with that repository. Try setting `/etc/apk/repositories` to `http://nl.alpinelinux.org/alpine/v3.3/main` then running `apk update; apk upgrade`.

**Comment by maier** [comment]
Actually, it looks like it may not even be an active repository any longer. Set the repositories file as above, then you can run `setup-apkrepos -f` to find the fastest repository available at the moment.

**Comment by samlevin** [comment]
I understand your comment and what to do, but how do I ensure it happens before NFS setup phase occurs? My `vagrant up` command fails at the NFS step, but the box becomes available. I SSH in, run the above (fine), and then leave to re-run `vagrant up` but it says `VirtualBox VM is already running.` Meanwhile, NFS is still broken.

**Comment by maier** [comment]
hmm. i _think_ i might be able to add a 'setup-apkrepos -f' into the nfs client install of the plugin so that it has a fresh list of repositories. let me test it and i'll report back.

**Comment by samlevin** [comment]
That would be ideal. Not to pester, but what is the concrete workaround? Do these actions myself, and then re-run `vagrant up --provision`?

**Comment by maier** [comment]
yeah, that would be a short-term workaround.

**Comment by samlevin** [comment]
Subscribed for updates on whether that can be added on your end due to the bunk repo link. Thanks for your help, and thanks for the great repo

**Comment by affixalex** [comment]
I built a derivative of this... https://github.com/hypoalex/packer-templates/tree/master/alpine - I just wanted it to upgrade to the latest edge release when it initially gets provisioned. The latest version of Alpine does include the guest utils, but the busybox mount command chokes on the syntax used to mount vboxfs filesystems. Here is a working example, not optimized for size though.

```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.configure(2) do |config|
  config.vm.box = "hypoalex/alpine"
  config.vm.provider "virtualbox"
  config.vm.network :private_network, ip: "172.168.65.2"
  config.vm.synced_folder ".", "/code", type: "nfs"

  config.vm.provision "shell", inline: <<-SHELL
    sudo apk update
    sudo apk upgrade
    sudo apk add alpine-sdk
    sudo apk add make
    sudo apk add gcc
    sudo apk add gperf
    sudo apk add vim
  SHELL
end
```

**Comment by samlevin** [comment]
@hypoalex that looks great. size optimization isn't important at this stage, so not a problem. Out of curiosity, have you had any luck installing docker on top of your fork?

**Comment by affixalex** [comment]
I haven't tried it, but I'd expect it to work. My fork defaults to using only 256MB of memory so you'd probably want to change that for Docker. I just wanted to do static builds against the latest musl-libc in the edge release. Halting and rebooting the machine seemed to cause issues, my fork isn't perfect but it works for my purposes. Getting this "right" would pretty much require building the full GNU `mount` command and including that, as well as just sticking to the latest stable release for the repositories used in the actual Packer templates (the edge release is used now).

**Comment by maier** [comment]
that is another alternative, using a box with a more up-to-date version of alpine. the issue with the plugin not being able to install the nfs client on the _specific_ v3.3 alpine virtualbox is that the box's apk repository configuration `/etc/apk/repositories` points to a repository that no longer exists. (repos.dfw.lax-noc.com)

**Comment by maier** [comment]
plugin 0.3.0 released and works with my alpine 3.3 boxes (which reference the defunct `repos.dfw.lax-noc.com` repo). let me know if you guys have any issues with the updated plugin.

**Comment by samlevin** [comment]
Latest image solved the repo problems and NFS works as expected. Thanks!
